src/test/scala/mesosphere/marathon/integration/LeaderIntegrationTest.scala

```diff
@@ -53,6 +58,110 @@ class LeaderIntegrationTest extends IntegrationFunSuite
     WaitTestSupport.waitUntil("the leader changes", 30.seconds) { marathon.leader().value != leader }
   }
 
+  test("it survives a small burn-in reelection test") {
```

Comment:
I like the idea of a burn-in reelection test. What about also checking that once an election happens, all all nodes return the same leader? There's an already existing test that does that, so you could re-use it.